cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -103,12 +103,15 @@ void setProvisioning(ProgramRunId id, long startTime, Map<String, String> runtim
    * Logs initialization of program run and persists program status to {@link ProgramRunStatus#STARTING}.
    *
    * @param id run id of the program
+   * @param startTime start timestamp in seconds
    * @param twillRunId Twill run id
+   * @param runtimeArgs the runtime arguments for this program run
    * @param systemArgs the system arguments for this program run
    * @param sourceId id of the source of program run status, which is proportional to the timestamp of
    *                 when the current program run status is reached
    */
-  void setStart(ProgramRunId id, @Nullable String twillRunId, Map<String, String> systemArgs, byte[] sourceId);
+  void setStart(ProgramRunId id, long startTime, @Nullable String twillRunId,
```

Comment:
How's startTime compare to the time in the run id? The RunId itself itself is a time-based UUID, which the timestamp is the time it was generated, which is considered as the program start time.